src/Ocelot/Configuration/Validator/FileConfigurationFluentValidator.cs

```diff
@@ -170,5 +171,52 @@ private static bool IsPlaceholderNotDuplicatedIn(string upstreamPathTemplate)
 
             return matchingRoutes.Count() <= 1;
         }
+
+        private static bool AreDuplicateUpstreamRoutingHeaders(FileRoute first, FileRoute second)
+        {
+            if (!first.UpstreamHeaderRoutingOptions.Headers.Any() && !second.UpstreamHeaderRoutingOptions.Headers.Any())
+            {
+                return true;
+            }
+
+            if (first.UpstreamHeaderRoutingOptions.Headers.Any() ^ second.UpstreamHeaderRoutingOptions.Headers.Any())

```

Comment:
Because if either of the two header collections is empty while the other is not, it's obvious that they can never be duplicate. See the method where the code is.

Reply:
Yeah, it should be a comment line, right? Like:
```csharp
// if either of the two header collections is empty while the other is not, it's obvious that they can never be duplicate
if (first.Any() ^ second.Any())
{ }
```

Reply:
Will add. :+1:

Reply:
✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1684/commits/dbe0f33344d1ff0bf1a356964ac8d59806ee68cd